src/braket/circuits/measure.py

```diff
@@ -0,0 +1,109 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+from __future__ import annotations
+
+from collections.abc import Sequence
+from typing import Any, Optional
+
+from braket.circuits.quantum_operator import QuantumOperator
+from braket.circuits.serialization import (
+    IRType,
+    OpenQASMSerializationProperties,
+    SerializationProperties,
+)
+from braket.registers.qubit_set import QubitSet
+
+
+class Measure(QuantumOperator):
+    """Class `Measure` represents a measure operation on targeted qubits"""
+
+    def __init__(
+        self, qubit_count: Optional[int] = 1, ascii_symbols: Sequence[str] = ["M"], **kwargs
+    ):
+        """Inits a `Measure`.
+
+        Args:
+            qubit_count (Optional[int]): Number of qubits this measure operation interacts with.
+            ascii_symbols (Sequence[str]): ASCII string symbols for the measure. These are used when
+                printing a diagram of circuits. Length must be the same as `qubit_count`, and
+                index ordering is expected to correlate with target ordering on the instruction.
+
+        Raises:
+            ValueError: `qubit_count` is less than 1, `ascii_symbols` are `None`, or
+                `ascii_symbols` length != `qubit_count`
+        """
+        super().__init__(qubit_count=qubit_count, ascii_symbols=ascii_symbols)
+        self._target_index = kwargs.get("index")
+
+    @property
+    def name(self) -> str:
+        return self.__class__.__name__
```

Comment:
I'm not sure I follow why we need this

Reply:
You are right. This is not used. Removed it.